packages/backend/src/api/endpoints/DomainApi.ts

```diff
@@ -0,0 +1,14 @@
+import { joinPaths } from '../../util/path';
+import type { Domains } from '../resources/Domains';
+import { AbstractAPI } from './AbstractApi';
+
+const basePath = '/domains';
+
+export class DomainAPI extends AbstractAPI {
+  public async deleteDomain(id: string) {
+    return this.request<Domains>({
```

Comment:
:question: Is `Domains` the return type here? If yes, it should be a [DeletedObject](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/resources/DeletedObject.ts).

Reply:
@gkats I'm a bit confused about this. What determines if we are going to use the `DeleteObject`? 

We seem to have it both ways. We seem to use it in [deletePhoneNumber](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/endpoints/PhoneNumberApi.ts#L50) and [deleteEmailAddress](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/endpoints/EmailAddressApi.ts#L50) but not in [deleteOrganization](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/endpoints/OrganizationApi.ts#LL129C16-L129C34), [deleteUser](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/endpoints/UserApi.ts#LL139C16-L139C26) and [disableUserMFA](https://github.com/clerkinc/javascript/blob/3be2c91a0e9119ade32cf195ba1cae2e97e1e74e/packages/backend/src/api/endpoints/UserApi.ts#LL163C16-L163C30) for example.

cc @dimkl @nikosdouvlis 
Can you clarify the thought behind this a bit more?

Reply:
Hey @raptisj, I can see why you might be confused by this. It's true that we're not consistent in our return types and we probably use some wrong types in certain functions. Sorry for that.

> What determines if we are going to use the DeleteObject

We basically have to check the API response. At the moment there's no documentation for the DELETE /v1/domains so you have every right to be confused. Here's an example for the [delete organization operation](https://clerk.com/docs/reference/backend-api/tag/Organizations#operation/DeleteOrganization). You'll notice the successful response body is a "DeletedObject".

So, again, apologies for not making that clear to you in the first place, but the DELETE /v1/domains endpoint will return a "DeletedObject" on successful invocation.

I'll update our docs shortly.